lib/codegen/analysis/layout.cc

```diff
@@ -451,7 +523,11 @@ void layouts::make_graph(ir::instruction *i) {
 void layouts::create(size_t id, const std::vector<ir::value*>& values) {
 //  if(layouts_.find(id) != layouts_.end())
 //    return;
-  auto it_hmma_c = std::find_if(values.begin(), values.end(), &is_hmma_c);
+  auto it_hmma_c = values.begin();
+  for (; it_hmma_c != values.end(); ++it_hmma_c) {
+    if (is_hmma_c(*it_hmma_c, tgt_->as_nvidia()->sm()))
+      break;
+  }
```

Comment:
What is the rationale for replacing the `find_if` here? Probably it can be retained (provided of course that the predicate is replaced accordingly)